### PR TITLE
use top level resources field in LaunchAssetLoaderResourceQuery

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
Lets us fetch the resources for the job without neccesarily needing to fetch the full job snapshot.

Test Plan: Launch an asset job in the UI

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx [hunk 1]
```diff
@@ -959,15 +931,7 @@ async function checkIfResourcesRequireConfig({
...
-  const pipeline = resourceResult.data.pipelineOrError;
-  if (pipeline.__typename !== 'Pipeline') {
-    return {
-      error: pipeline.message,
-      requiredConfigPresent: false,
-    };
-  }
-
-  const resources = pipeline.modes[0]!.resources.filter((r) =>
+  const resources = resourceResult.data.resources.filter((r) =>
```
Comment (alangenfeld) [inline js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx:934]: what happens here if we hit some weird desync and the job selector misses ? I worry that if `resources` is not in `data` (or `data` not in result) the JS error will propagate and blow up page 

 re comment on other PR if its nullable we can trigger the error condition [1] when we get back null resources 
 
 going the whole way and doing an `OrError` probably overkill but worth considering

Comment (alangenfeld) [reply js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx:934]: we unfortunately do not always use the implicit asset job

https://github.com/dagster-io/dagster/blob/master/js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx#L493

https://github.com/dagster-io/dagster/blob/master/js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx#L601-L608

reason i have this cursed knowledge https://github.com/dagster-io/dagster/pull/25772

Comment (gibsondan) [reply js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx:934]: the second comment is interesting cursed knowledge but I don't totally follow how it relates to this PR

Comment (alangenfeld) [reply js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx:934]: second comment is just context for why i think desync concern raised in first comment is possible enough to have to care about

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**